### PR TITLE
Basic `vector` property tests

### PR DESCRIPTION
**Pull request by dmalkr**
Just copy `vector` package property tests

### CODE REVIEW

File: vecvec-lapack/vecvec-lapack.cabal [hunk 1]
```diff
@@ -44,20 +44,33 @@ Library
...
                   TST.Slice
                   TST.Orphanage
+                  TST.Property
+  extensions:     CPP
+                  FlexibleContexts
+                  ImportQualifiedPost
+                  RankNTypes
+                  ScopedTypeVariables
+                  TemplateHaskell
+                  TypeFamilies
```
Comment (Shimuuar) [inline vecvec-lapack/vecvec-lapack.cabal]: Please enable CPP & TH on per module basis

**Review by Shimuuar — CHANGES_REQUESTED** [review]
Great! Thanks

There's quite a bit of orphans we don't need: all arbitrary instances for vectors from `vector`. Please kill unused ones and move remaining ones into `TST.Orphanage`. And drop `-fno-orphans` from cabal file

Please also make code warning free. `TST.Property` has tons of top-level functions without signature so just drop `{-# OPTIONS_GHC -Wno-missing-signatures #-}` there

**Comment by Shimuuar** [comment]
Oh! Another thing. Please add possibility to generate vectors with non-1 stride in Arbitrary for Vec. See following as an example:

https://github.com/Shimuuar/vecvec/blob/7d73f0498328fad5a63f88762ef478feac179354/vecvec-lapack/tests/TST/Model.hs#L345-L348

File: vecvec-lapack/vecvec-lapack.cabal (hunk 1 above)
Comment (dmalkr) [reply vecvec-lapack/vecvec-lapack.cabal]: CPP & TH moved to modules.

**Comment by dmalkr** [comment]
> Oh! Another thing. Please add possibility to generate vectors with non-1 stride in Arbitrary for Vec. See following as an example:
> 
> https://github.com/Shimuuar/vecvec/blob/7d73f0498328fad5a63f88762ef478feac179354/vecvec-lapack/tests/TST/Model.hs#L345-L348

**Comment by dmalkr** [comment]
Oh, got confused in the comment interface and accidentally closed the PR. :(

**Comment by dmalkr** [comment]
> Oh! Another thing. Please add possibility to generate vectors with non-1 stride in Arbitrary for Vec. See following as an example:
> 
> https://github.com/Shimuuar/vecvec/blob/7d73f0498328fad5a63f88762ef478feac179354/vecvec-lapack/tests/TST/Model.hs#L345-L348

Done for `Arbitrary`. `TestData` also need to make `Vec`s from lists, but it have no access to random generating. So I made here fixed stride:

https://github.com/Shimuuar/vecvec/blob/e0fe928708bfa55e1f96a6a60ad8a2f7ab5b7963/vecvec-lapack/tests/Utilities.hs#L114-L120

File: vecvec-lapack/vecvec-lapack.cabal (hunk 1 above)
Comment (dmalkr) [reply vecvec-lapack/vecvec-lapack.cabal]: Both files, `Utilities.hs` and `TST/Property.hs` are copied from `vector`. And we hope tests from `vector` will be separated into other package. So we can just import them. So I didn't move any orphans, and just silent such warnings.

**Pull request merged**